webauthn4j-core/src/main/java/com/webauthn4j/validator/attestation/statement/tpm/DefaultTPMDevicePropertyDecoder.java

```diff
@@ -0,0 +1,70 @@
+package com.webauthn4j.validator.attestation.statement.tpm;
+
+import com.webauthn4j.validator.exception.BadAttestationStatementException;
+import org.apache.kerby.asn1.type.Asn1Utf8String;
+
+import javax.naming.InvalidNameException;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attributes;
+import javax.naming.ldap.LdapName;
+import javax.naming.ldap.Rdn;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DefaultTPMDevicePropertyDecoder implements TPMDevicePropertyDecoder {
+
+    public TPMDeviceProperty decode(String subjectAlternativeName) throws BadAttestationStatementException {
+        try{
+            LdapName subjectDN;
+            try {
+                subjectDN = new LdapName(subjectAlternativeName);
+            } catch (InvalidNameException e) {
+                throw new IllegalArgumentException(e);
+            }
+            Map<String, Object> map = subjectDN.getRdns().stream().flatMap(rdn -> convertRdnToMap(rdn).entrySet().stream()).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+            byte[] manufacturerAttr = (byte[]) map.get("2.23.133.2.1");
+            byte[] partNumberAttr = (byte[]) map.get("2.23.133.2.2");
+            byte[] firmwareVersionAttr = (byte[]) map.get("2.23.133.2.3");
+
+            String manufacturer = decodeAttr(manufacturerAttr);
+            String partNumber = decodeAttr(partNumberAttr);
+            String firmwareVersion = decodeAttr(firmwareVersionAttr);
+            return new TPMDeviceProperty(manufacturer, partNumber, firmwareVersion);
+        }
+        catch (IOException e) {
+            throw new BadAttestationStatementException("The Subject Alternative Name extension of attestation certificate does not contain a TPM device property", e);
+        }
+    }
+
+    String decodeAttr(byte[] attr) throws IOException {
```

Comment:
nit: probably a good candidate for static utility method?

Reply:
Yes I think it should do it for our use case, thanks for taking care of it quickly